.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: Build
+
+on: [ push ]
+
+jobs:
+  lint_and_format:
+    name: Lint and format checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Ruff lint
+        uses: astral-sh/ruff-action@v3
+        with:
+          args: check --output-format=github
+
+      - name: Ruff format
+        uses: astral-sh/ruff-action@v3
+        with:
+          args: format --check
+
+  run_tests:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+
+      - name: Install uv
+        run: |
+          pip install uv
+          uv sync --all-groups
+
+      - name: Run pytest
+        run: uv run pytest
```

Comment:
Added another step to run the tests. 

1. Checks out the code
2. Sets up python (based on python version in `.python-version`)
3. Installs uv, uses uv to install python packages. `--all-groups` pulls in what's defined in the dev group.
4. Runs pytest.